### PR TITLE
feat: implement comprehensive webhook integration with thirdweb

### PR DESCRIPTION
**Pull request by chibie**
### Description

This PR implements comprehensive webhook integration with thirdweb's Insight Webhook API for real-time blockchain event processing. The implementation replaces polling-based event detection with real-time webhooks, enhancing security and performance.

**Key Features:**
- Smart webhook management with automatic creation/updates based on network configuration
- Real-time blockchain event processing for Transfer, OrderCreated, OrderSettled, and OrderRefunded events
- HMAC-SHA256 signature verification for webhook security
- Encrypted recipient data storage in `message_hash` field
- Automatic webhook cleanup when lock payment orders are created
- Server startup integration for automatic webhook setup

**Database Changes:**
- New `PaymentWebhook` schema with one-to-one relationship to Network
- `message_hash` field added to PaymentOrder (400 chars for RSA-2048 ciphertext)
- Proper relationships between Network ↔ PaymentWebhook and PaymentOrder ↔ PaymentWebhook

**API Changes:**
- New `/v1/insight/webhook` endpoint for processing thirdweb webhook events
- Updated order processing to use `message_hash` instead of `tx_hash` for uniqueness
- Removed `WaitForTransactionMined` calls from order management functions

**Security Enhancements:**
- Webhook signature verification with HMAC-SHA256
- Payload age verification (5-minute window)
- Address normalization for consistency
- Encrypted recipient data storage

### References

- Implements thirdweb Insight Webhook API integration
- Replaces polling-based event detection with real-time webhooks
- Enhances payment order processing with encrypted recipient data

### Testing

**Unit Tests:**
- Added tests for `CreateGatewayWebhook`, `GetWebhookByID`, and `UpdateWebhook` functions
- Added tests for webhook endpoint signature verification and event processing
- Tests cover error scenarios and edge cases

**Integration Tests:**
- Webhook endpoint testing with valid/invalid signatures
- Event processing for Transfer and OrderCreated events
- Database consistency verification

**Manual Testing:**
- Gateway webhook creation on server startup
- Webhook update when new networks are added
- End-to-end payment flow with webhook events
- Error handling for various failure scenarios

**Environment:**
- Developed on macOS with Go 1.23.1
- Tested with thirdweb Insight API
- Database: PostgreSQL with ent ORM

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation and tests for new/changed/fixed functionality in this PR
- [ ] All active GitHub checks for tests, formatting, and security are passing (some existing tests fail due to unrelated issues)
- [x] The correct base branch is being used (stable)

By submitting a PR, I agree to Paycrest's [Contributor Code of Conduct](https://paycrest.notion.site/Contributor-Code-of-Conduct-1602482d45a2806bab75fd314b381f4c) and [Contribution Guide](https://paycrest.notion.site/Contribution-Guide-1602482d45a2809a8930e6ad565c906a).

### CODE REVIEW

File: controllers/index.go [hunk 1]
```diff
@@ -1054,3 +1061,365 @@ func (ctrl *Controller) IndexTransaction(ctx *gin.Context) {
...
+	if strings.EqualFold(fromAddress, token.Edges.Network.GatewayContractAddress) {
+		return nil
+	}
+
+	// Parse transfer value
+	transferValue, err := decimal.NewFromString(valueStr)
+	if err != nil {
+		return fmt.Errorf("invalid transfer value: %w", err)
+	}
+
```
[the remark this answers is not in the document]
Comment (chibie) [reply controllers/index.go:1255]: minimum check should rather happen during order initiation
